client/shipping-label/views/index.js

```diff
@@ -1,112 +1,222 @@
-import React, { PropTypes } from 'react';
+import ReactDom from 'react-dom';
+import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import Button from 'components/button';
+import CalypsoI18n from 'i18n-calypso';
 import { translate as __ } from 'lib/mixins/i18n';
+import LabelHistory from './history';
 import PurchaseLabelDialog from './purchase';
 import RefundDialog from './refund';
 import ReprintDialog from './reprint';
 import TrackingLink from './tracking-link';
 import Spinner from 'components/spinner';
+import Gridicon from 'components/gridicon';
+import Tooltip from 'components/tooltip';
 import formatDate from 'lib/utils/format-date';
 import * as ShippingLabelActions from 'shipping-label/state/actions';
 import notices from 'notices';
 import GlobalNotices from 'components/global-notices';
 import getFormErrors from 'shipping-label/state/selectors/errors';
 import canPurchase from 'shipping-label/state/selectors/can-purchase';
 import _ from 'lodash';
+import { sprintf } from 'sprintf-js';
 
-let needToFetchLabelsStatus = true;
+class ShippingLabelRootView extends Component {
+	constructor( props ) {
+		super( props );
 
-const ShippingLabelRootView = ( props ) => {
-	const renderPurchaseLabelFlow = () => {
+		this.renderLabel = this.renderLabel.bind( this );
+		this.openTooltip = this.openTooltip.bind( this );
+		this.closeTooltip = this.closeTooltip.bind( this );
+
+		this.needToFetchLabelsStatus = true;
+
+		this.state = {
+			showTooltips: props.shippingLabel.labels.map( () => false ),
+		};
+	}
+
+	componentDidMount() {
+		this.historyMetaBox = document.getElementById( 'wc-connect-shipping-label-history' );
```

Comment:
The other meta-box should be another completely different React "application", but currently we rely on a global variable `wcConnectData` so it's not possible. Tomorrow I have a ✈️ , I'll give this a shot and see if I can make 2 applications live in the same page.